File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int currMax = arr[arr.size() - 1];
        for (int i = arr.size() - 2; i > -1; --i) {
            if (arr[i] > currMax) {
                int prevMax = currMax;
                currMax = arr[i];
                arr[i] = prevMax;
            }
            else arr[i] = currMax;
        }
        arr[arr.size() - 1] = -1;
        return arr;
    }
};

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        greatest = arr[-1]
        arr[-1] = -1

        for i in range(len(arr) - 2, -1, -1):
            curr = arr[i]
            arr[i] = greatest
            greatest = max(curr, greatest)

        return arr